internal/pkg/service/stream/storage/level/local/diskwriter/network/transport/transport_test.go

```diff
@@ -0,0 +1,108 @@
+package transport_test
+
+import (
+	"context"
+	"io"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/yamux"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	commonDeps "github.com/keboola/keboola-as-code/internal/pkg/service/common/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/diskwriter/network"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/diskwriter/network/transport"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+func TestServerAndClient(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
```

Comment:
Implementing a graceful shutdown and a stable test was quite difficult.

In the future, it would be good to add more tests, using `toxiproxy` and test e.g. keep alive mechanism.

We are already using the `toxiproxy` for some critical etcd tests, to simulate network outage.
https://github.com/keboola/keboola-as-code/blob/484f05099bf157c0cf92330da5c21c5d547b382e/internal/pkg/service/common/etcdop/session_test.go#L10